src/main/scala-2.11/markets/clearing/ClearingMechanismLike.scala

```diff
@@ -15,13 +15,45 @@ limitations under the License.
 */
 package markets.clearing
 
-import markets.BaseActor
+import akka.actor.ActorRef
+import akka.agent.Agent
+
+import markets.{BaseActor, Cancel, Canceled}
 import markets.clearing.engines.MatchingEngine
+import markets.orders.Order
+import markets.tickers.Tick
 
 
+/** Mixin Trait providing ClearingMechanismLike behavior to some BaseActor. */
 trait ClearingMechanismLike {
   this: BaseActor =>
 
-  protected[this] val matchingEngine: MatchingEngine
+  def matchingEngine: MatchingEngine
+
+  def settlementMechanism: ActorRef
+
+  def ticker: Agent[Tick]
+
+  def clearingMechanismBehavior: Receive = {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/1560410920/issues/source?bid=2921361&fileBranchId=3019109#l37)